### PR TITLE
Add antsibull_core.yaml.store_yaml_stream() 

### PR DESCRIPTION
**Pull request by gotmax23**

### CODE REVIEW

File: src/antsibull_core/yaml.py [hunk 1]
```diff
@@ -43,11 +43,12 @@ def load_yaml_file(path: StrOrBytesPath) -> t.Any:
         return yaml.load(stream, Loader=_SafeLoader)
 
 
-def store_yaml_file(path: StrOrBytesPath, content: t.Any) -> None:
+def store_yaml_file(path: StrOrBytesPath, content: t.Any, append: bool = False) -> None:
```
Comment (felixfontein) [inline src/antsibull_core/yaml.py:46]: Hmm, I don't really like the approach of writing a file and then appending later on, especially if it happens in the same calling function. Code really should open a file for writing once and then write all content it wants to write instead.

What do you think of the following approaches:
1. Have a function that stores YAML into `bytes`;
2. Have a function that writes YAML into a files object;
3. Or both?

Comment (gotmax23) [reply src/antsibull_core/yaml.py]: Yeah, you're exactly right. I've added a `store_yaml_stream` function to write YAML into a file object.

File: changelogs/fragments/24.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+minor_changes:
+  - Add a ``store_yaml_stream`` function to ``antsibull_core.yaml`` to dump
+    YAML to an IO stream.
```
Comment (felixfontein) [inline changelogs/fragments/24.yaml:4]: You seem to like these periods before the final brackets :)
```suggestion
    YAML to an IO stream
```

**Review by felixfontein — COMMENTED** [review]
Besides this, LGTM.

File: changelogs/fragments/24.yaml (hunk 2 above)
Comment (gotmax23) [reply changelogs/fragments/24.yaml:4]: It seems I do :)

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@gotmax23 thanks!

**Pull request merged**